src/browser.js

```diff
@@ -40,29 +38,38 @@ class UniversalEmitter extends Emitter {
   emit(type: mixed, payload: mixed): void {
     payload = super.mapEvent(type, payload);
     super.handleEvent(type, payload);
-    this.batch.push({type, payload});
+    add({type, payload});
   }
   // match server api
   from(): UniversalEmitter {
     return this;
   }
-  flushInternal(): void {
-    if (this.batch.length > 0) {
-      this.fetch('/_events', {
+  async flushInternal(): Promise<void> {
+    const items = getAndClear();
```

Comment:
Wonder why are we clearing before ensuring the batch is sent. 
`empty -> send -> add back if fail` vs `send -> clear if success` seems like the latter is safer?

Reply:
If the request takes a really long time, new events could have been added to the list - so its unsafe just to empty the array at the end.